pandas/core/algorithms.py

```diff
@@ -1573,9 +1580,10 @@ def searchsorted(arr, value, side="left", sorter=None) -> np.ndarray:
             dtype = value_arr.dtype
 
         if is_scalar(value):
-            value = dtype.type(value)
+            # We know that value is int
+            value = cast(int, dtype.type(value))
         else:
-            value = pd_array(value, dtype=dtype)
+            value = pd_array(cast(ArrayLike, value), dtype=dtype)
```

Comment:
if we know its ArrayLike then why do we need pd_array?

Reply:
could be a change in dtype

Reply:
then an astype would be preferable wouldnt it?  also would retain ndarray instead of casting to PandasArray, which IIUC would just be unpacked in the arr.searchsorted call